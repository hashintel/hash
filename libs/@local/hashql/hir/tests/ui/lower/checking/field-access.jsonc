//@ run: skip reason="structs and tuple definitions are disabled for now"
//@ description: Field access inference
["let", "foo", { "#struct": { "a": { "#literal": 42 } } }, "foo.a"]
