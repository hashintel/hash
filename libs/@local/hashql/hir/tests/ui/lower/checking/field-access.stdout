════ Initial HIR ═══════════════════════════════════════════════════════════════

#let foo:0 = (a: 42) in
foo:0.a

════ HIR after type checking ═══════════════════════════════════════════════════

#let foo:0 = (a: 42) in
foo:0.a

════ Types ═════════════════════════════════════════════════════════════════════

┌─ 42
└→ Integer

┌─ (a: 42)
└→ (a: Integer)

┌─ foo:0
└→ (a: Integer)

┌─ foo:0.a
└→ Integer

┌─ #let foo:0 = (a: 42) in
│  foo:0.a
└→ Integer

